src/oscar/apps/catalogue/abstract_models.py

```diff
@@ -115,6 +115,19 @@ def full_name(self):
         names = [category.name for category in self.get_ancestors_and_self()]
         return self._full_name_separator.join(names)
 
+    def get_full_slug(self, parent_slug=None):
+        if self.is_root():
+            return self.slug
+
+        cache_key = self.get_url_cache_key()
```

Comment:
`get_url_cache_key()` was used to cache the **URL** of the category. It is now being used to cache the slug, which is not the same thing, and will break projects that have overridden any of this behaviour, e.g., to use a different URL structure. I think we should stick to caching the URL as before, and if there is some additional benefit then caching the slug in a separate key.

Reply:
Don't you think it is incredibly _weird_ to have a different method of operation for get_absolute_url and reverse? I think caching the full url is a mistake hereby corrected. Caching the full_slug, amounts to the same speed, but leaves operation of reverse the same as operation of get_absolute_url. Please note that this is also the way it was before. full_slug used to be a property of the product which was removed by normalization. get_url_cache_key merely calculates the **unique key** under which some value is stored in the cache, why would this have any effect on custom url structure?